woocommerce/woocommerce-gutenberg-products-block
# Rating Product block: Add support for global style

~This PR is blocked by #5515~

This PR adds support for global style for the `Rating Product` block.

Now, the user can edit the style for:
- text-color

Fixes #4965 

### Screenshots

![image](https://user-images.githubusercontent.com/4463174/148258248-05e71cfe-c0c9-459e-955b-d986b15fd872.png)
*without global style*


![image](https://user-images.githubusercontent.com/4463174/148258176-1cef6241-930d-422c-8e36-caa1f7429234.png)
*with global style*


### Testing

### Manual Testing

Check out this branch:

1. On `WordPress 5.9`, install and enable the `Gutenberg` plugin.
2. Install and enable the `Twenty Twenty-Two` theme.
3. Add the `Single Product block` (this block contains `Rating Product block`) to a post.
4. Get the focus on `Rating Product block`.
5. On the right sidebar, personalize the styles of the block.
6. Go on the page and check if there are changes.
7. Reset to default using the `Reset` button from the different sections. 
8. Go to Dashboard and select Appearance > Editor (beta). On top of the screen, select Home > Browser all templates > Single Post. When the page is loaded, add the block to the page.
9. On the Editor page click on the `Styles` icon on the right-top corner.
10. Verify that the `Rating Product block` is shown under the `Blocks` section. Personalize again the block.
11. Save your changes.
12. Go on the page created earlier and check if all styles are applied correctly.
13. Edit your previous post/page again.
14. Change again the styles.
15. Save your changes.
16. Check if these styles have priority over the styles from the Site Editor.

### Changelog

> Add support for the global style for the Rating Product block.
